graylog2-server/src/main/java/org/graylog2/indexer/IndexSet.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.indexer;
+
+import org.graylog2.indexer.indices.TooManyAliasesException;
+
+import java.util.Map;
+import java.util.Set;
+
+public interface IndexSet {
+    String[] getAllGraylogIndexNames();
```

Comment:
Does this method return *all* Graylog-managed index names, including those not in this index set? If not, the method name might be misleading.

The return value could also be a `Set<String>` to make sure that each index name is only included once.